### PR TITLE
Convert grader-host to TypeScript

### PR DESCRIPTION
**Pull request by nwalters512**
Depends on #10049.

### CODE REVIEW

**Review by mylesw27 — APPROVED** [review]
The TS conversions look correct. I am unable to test locally but do not want to block on that.

**Comment by nwalters512** [comment]
I'll test this in our staging environment before merging.

File: apps/grader-host/src/index.ts
Comment (jonatanschroeder) [inline apps/grader-host/src/index.ts:1]:
```suggestion
import type { Readable } from 'node:stream';
```

File: apps/grader-host/src/index.ts
Comment (jonatanschroeder) [inline apps/grader-host/src/index.ts:217]: Do you want to import this?

File: apps/grader-host/src/lib/receiveFromQueue.ts
Comment (jonatanschroeder) [inline apps/grader-host/src/lib/receiveFromQueue.ts:7]:
```suggestion
  type SQSClient,
```

File: apps/grader-host/src/lib/receiveFromQueue.ts
Comment (jonatanschroeder) [inline apps/grader-host/src/lib/receiveFromQueue.ts:89]: Did you mean `break`?

**Review by jonatanschroeder — APPROVED** [review]

File: apps/grader-host/src/index.ts
Comment (nwalters512) [reply apps/grader-host/src/index.ts:217]: Yep - this was a bad merge conflict resolution.

File: apps/grader-host/src/lib/receiveFromQueue.ts
Comment (nwalters512) [reply apps/grader-host/src/lib/receiveFromQueue.ts]: Done in d7e7403.

File: apps/grader-host/src/index.ts
Comment (nwalters512) [reply apps/grader-host/src/index.ts]: Fixed in d7e7403.

File: apps/grader-host/src/index.ts
Comment (nwalters512) [reply apps/grader-host/src/index.ts]: Done in d7e7403.

**Pull request merged**
